src/memory_manager.rs

```diff
@@ -287,13 +397,96 @@ impl<M: Memory> MemoryManagerInner<M> {
             bucket_size_in_pages: header.bucket_size_in_pages,
             memory_sizes_in_pages: header.memory_sizes_in_pages,
             memory_buckets,
+            freed_buckets: BTreeSet::new(),
+        }
+    }
+
+    fn load_layout_v2(memory: M, header: Header) -> Self {
```

Comment:
I've been finding this function generally difficult to read. I'll add some pointers below on how to make it more readable.

Reply:
Done. PTAL.